controllers/component_controller.go

```diff
@@ -470,46 +480,51 @@ func (r *ComponentReconciler) generateGitops(ctx context.Context, req ctrl.Reque
 	}
 
-	// Create a temp folder to create the gitops resources in
-	tempDir, err := ioutils.CreateTempPath(component.Name, r.AppFS)
-	if err != nil {
-		log.Error(err, "unable to create temp directory for GitOps resources due to error")
-		return fmt.Errorf("unable to create temp directory for GitOps resources due to error: %v", err)
-	}
-
-	deployAssociatedComponents, err := devfileParser.GetDeployComponents(compDevfileData)
-	if err != nil {
-		log.Error(err, "unable to get deploy components")
-		return err
-	}
+	// Determine if we're using a Kubernetes job for gitops generation, or generating locally
+	localGitopsGen := (r.AllowLocalGitopsGen && component.Annotations["allowLocalGitopsGen"] == "true") || (r.DoLocalGitOpsGen)
+	if localGitopsGen {
+		err = gitopsjoblib.GenerateGitopsBase(context.Background(), r.Client, *component, ioutils.NewFilesystem(), generate.GitOpsGenParams{
+			Generator: r.Generator,
+			Token:     r.GitToken,
+			RemoteURL: gitOpsURL,
+			Branch:    gitOpsBranch,
+			Context:   gitOpsContext,
+		})
+		if err != nil {
+			return err
+		}
+	} else {
+		gitopsJobConfig := gitopsjob.GitOpsJobConfig{
+			OperationType: gitopsjob.GenerateBase,
+			RepositoryURL: component.Status.GitOps.RepositoryURL,
+			ResourceName:  component.GetName(),
+			Branch:        gitOpsBranch,
+			Context:       gitOpsContext,
+		}
 
-	kubernetesResources, err := devfile.GetResourceFromDevfile(log, compDevfileData, deployAssociatedComponents, component.Name, component.Spec.ContainerImage)
-	if err != nil {
-		log.Error(err, "unable to get kubernetes resources from the devfile outerloop components")
-		return err
-	}
+		// Create a Kubernetes job (with random string appended to name to ensure uniqueness) to generate the GitOps resources
+		jobName := component.GetName()
+		if len(jobName) > 57 {
+			jobName = component.GetName()[0:56]
+		}
+		jobName = jobName + util.GetRandomString(5, true)
+		jobNamespace := r.GitOpsJobNamespace
+		if jobNamespace == "" {
+			jobNamespace = component.Namespace
+		}
+		err = gitopsjob.CreateGitOpsJob(ctx, r.Client, r.GitToken, jobName, jobNamespace, component.Namespace, gitopsJobConfig)
+		if err != nil {
+			return err
+		}
 
-	// Generate and push the gitops resources
-	gitopsConfig := prepare.PrepareGitopsConfig(ctx, r.Client, *component)
-	mappedGitOpsComponent := util.GetMappedGitOpsComponent(*component, kubernetesResources)
+		// Wait for the job to succeed, error out if the 5 min timeout is reached
+		err = gitopsjob.WaitForJob(log, context.Background(), r.Client, r.GitOpsJobClientSet, jobName, jobNamespace, 5*time.Minute)
+		if err != nil {
+			return r.CleanUpJobAndReturn(log, jobName, jobNamespace, err)
+		}
 
-	err = r.Generator.CloneGenerateAndPush(tempDir, gitOpsURL, mappedGitOpsComponent, r.AppFS, gitOpsBranch, gitOpsContext, false)
-	if err != nil {
-		log.Error(err, "unable to generate gitops resources due to error")
-		return err
-	}
+		r.CleanUpJobAndReturn(log, jobName, jobNamespace, nil)
```

Comment:
```suggestion
		_ = r.CleanUpJobAndReturn(log, jobName, jobNamespace, nil)
```

otherwise you might get a gosec err if the PR CI all run without an issue